### ISSUE TITLE
Domain name not set correctly

### ISSUE THREAD

**Issue by gjelsas** [nextcloud/social#995]
In my profile page my social name is shown as @{name}@server server is the LAN-name of the machine Nextcloud is running on. it should be @{name}@fqdn.tld 

I also can't follow anybody right now. 

BTW: I recently changed the domain name and host machine of this installation and imported the database to the new host, but otherwise my installation seems okay.

**Comment by JimmyKater** [comment]
I didn't change my domainname at all and see the same problem on my NC19.0.2 instance: it uses the internal IP-address rather than the fqdn

(if you'd need more infos about my instance, feel free to ask, I'm gonna provide details asap

**Comment by violoncelloCH** [comment]
what do you have specified in your `config.php` for `overwrite.cli.url`?

**Comment by JimmyKater** [comment]
of course `https://my.fgdn.tld/`

**Comment by daita** [comment]
Yes, there is currently no way to switch domain name and keep your data from the social app. You will need to run `./occ social:reset` to reset and specify your new domainname at the end of the process.

Is everything solved in this ticket ?

**Comment by JimmyKater** [comment]
> Is everything solved in this ticket ?

I'm afraid, not! The problem was - at least what I was understanding from it - that the setupprocedure that was delivered with NC19.x automatically picked internal IP instead of registered FQDN... Which is clearly a mistake.

**Comment by violoncelloCH** [comment]
it's chosen from either 'overwite.cli.url' (at the point of setup) or the domain it's accessed over the first time (AFAIK - @daita can you confirm or point me to the respective code which determines the Domain used on set up?)

**Comment by daita** [comment]
[Yes](https://github.com/nextcloud/social/blob/8965acea039fc93b41f31d104c1b7330f11905df/lib/Controller/NavigationController.php#L204-L217), if the Social app is not configured, it will use the value from `overwrite.cli.url`

**Comment by daita** [comment]
please note that I am totally open to use another way to initialize this variable

**Comment by JimmyKater** [comment]
well in my case it didn't work like that... since `overwrite.cli.url` was (and is!) pointing to the correct fqdn...

the internal IP (which was picked) is only mentioned (at least with my setup) within the trusted hosts array.

**Comment by violoncelloCH** [comment]
so maybe we could also check the domain social is currently accessed over and compare with overwrite.cli.url...

> well in my case it didn't work like that... since overwrite.cli.url was (and is!) pointing to the correct fqdn...

strange, looking at the code that's hard to believe... can you check if you can reproduce this behaviour?

**Comment by JimmyKater** [comment]
> can you check if you can reproduce this behaviour?
ummm... as I can't update again to NC19 how else would that be possible? 

But I need to be honest... in NC18.0.x social was apparently unusable and I didn't check it anymore. So maybe the disconfiguration happened before? I can't be sure.  Plus I can't remember anymore if social was one of the apps which made a problem right after updating as there had been some I needed to manually re-activate again... 

Anyways. I could only describe what happened to me... I dunno why. overwrite.cli.url was set right and remained unchanged for years now. And I could reset that successfully by `./occ social:reset` - so at least the name@address would show right now.

**Comment by SchBenedikt** [comment]
Automatically closed — this is a configuration issue on an old version. The domain name is read from config, and if it's incorrect, it should be updated in Nextcloud's `config.php` or the Social settings.
